### ISSUE TITLE
problem processing wsdl Extension Elements

### ISSUE THREAD

**Issue by mgorczyca** [vpulim/node-soap#565]
To process my wsdl files, I needed to change the order of the desc and base parameters in the call to extend in ExtensionElement in wsdl.js. Before the change,  a string was passed where an obj was expected.

Code below, with original and changed line:
ExtensionElement.prototype.description = function(definitions, xmlns) {
  var children = this.children;
  var desc = {};
  for (var i=0, child; child=children[i]; i++) {
    if (child instanceof SequenceElement ||
      child instanceof ChoiceElement) {
      desc = child.description(definitions);
    }
  }
  if (this.$base) {
    var type = splitNSName(this.$base),
      typeName = type.name,
      ns = xmlns && xmlns[type.namespace] || definitions.xmlns[type.namespace],
      schema = definitions.schemas[ns];

    if (typeName in Primitives) {
      return this.$base;
    }
    else {
      var typeElement = schema && ( schema.complexTypes[typeName] ||
        schema.types[typeName] || schema.elements[typeName] );

      if (typeElement) {
        var base = typeElement.description(definitions, xmlns);
        //extend(desc, base);  // orig
        extend(base, desc);    // changed
      }
    }
  }
  return desc;
};

**Comment by hereblur** [comment]
This happened to me as well. When extend complexType from Primitive type. This is my work around, not sure if it's the best way.

```
var base = typeElement.description(definitions, xmlns);
        
if(typeof base=='string'){   
     return base;
}
          
extend(desc, base);
```

**Comment by balintbako** [comment]
Had similar problems while using a simpleType as  baseType in a complexType, however it worked when I used xs:string as baseType.
Not working wsdl part:
```
<xs:complexType name="EngineType">
	<xs:simpleContent>
		<xs:extension base="EngineEnumType">
			<xs:attribute name="Flatten" type="xs:boolean"/>
			<xs:attribute name="Intensity" type="qas:EngineIntensityType"/>
			<xs:attribute name="PromptSet" type="qas:PromptSetType"/>
			<xs:attribute name="Threshold" type="qas:ThresholdType"/>
			<xs:attribute name="Timeout" type="qas:TimeoutType"/>
		</xs:extension>
	</xs:simpleContent>
</xs:complexType>
<xs:simpleType name="EngineEnumType">
	<xs:restriction base="xs:string">
		<xs:enumeration value="Singleline"/>
		<xs:enumeration value="Typedown"/>
		<xs:enumeration value="Verification"/>
		<xs:enumeration value="Keyfinder"/>
	</xs:restriction>
</xs:simpleType>
```

The working wsdl part:
```
<xs:complexType name="EngineType">
	<xs:simpleContent>
		<xs:extension base="xs:string">
			<xs:attribute name="Flatten" type="xs:boolean"/>
			<xs:attribute name="Intensity" type="qas:EngineIntensityType"/>
			<xs:attribute name="PromptSet" type="qas:PromptSetType"/>
			<xs:attribute name="Threshold" type="qas:ThresholdType"/>
			<xs:attribute name="Timeout" type="qas:TimeoutType"/>
		</xs:extension>
	</xs:simpleContent>
</xs:complexType>
```

**Comment by mgorczyca** [comment]
I like the solution from hereblur, because it returns the base string as the description. Regarding balintbako's comment about using string for the base, I'm not clear as to how that solves the problem. To me, it looks like the problem is that the code expects base to be an object, so it does not work when base is a string. Which is why a fix/workaround is needed. Also, in my case, the wsdl is from a vendor so the goal is to use it 'as is'.

**Comment by herom** [comment]
A fix for this is highly apprecitated - it would be awesome if you could provide a working fix in a PR :+1:

**Comment by mgorczyca** [comment]
Have created the pull request, using the fix proposed by hereblur.

**Comment by mgorczyca** [comment]
Cleaned up the PR and resubmitted.

**Comment by hereblur** [comment]
Great, I'm glad it helped :)

**Comment by mgorczyca** [comment]
The fix for #585 also solves this issue for me.

**Comment by herom** [comment]
Thanks for the reply @mgorczyca 

I'm closing this as #589 is probably the fix for this issue. Please feel free to reopen if this problem is still there.
